### PR TITLE
Fix no clause match when key is an integer

### PR DESCRIPTION
**Pull request by williamthome**
For example `#{0 => 0}` throws no function clause matching.
This PR fixes this by escaping the key.
```console
1> thoas:encode(#{0 => 0}).
<<"{\"0\":0}">>
```
Maybe the output result should be <<"{0:0}">>?

### CODE REVIEW

**Comment by michalmuskala** [comment]
`{0:0}` is not valid JSON, it has to be `{"0":0}`. Keys are only allowed to be strings

**Comment by williamthome** [comment]
> `{0:0}` is not valid JSON, it has to be `{"0":0}`. Keys are only allowed to be strings 

Ah, sure, coding late at night I missed that. Thanks @michalmuskala

**Review by lpil — COMMENTED** [review]
Thank you @williamthome ! Could we get a test for this and also an entry in the change log file please?

**Comment by williamthome** [comment]
> Thank you @williamthome ! Could we get a test for this and also an entry in the change log file please?

Yes. I will turn this PR into a draft and set it ready for review when finished.

**Comment by williamthome** [comment]
@lpil done, but I have some two doubts.
The first one is the version. I set it to `v0.3.1` in the change log file. I see this as a minor change. Or will it be `v0.4.0`?
The second is the version in the mix project. What's the `@version "1.1.2"` related?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v0.3.1 - 2022-08-24
+
+- Fix error when encoding integer key.
```
Comment (lpil) [inline CHANGELOG.md]: This was not an error but instead it was something that was not supported. Could you change this to describe what the new feature is please? Thank you

**Review by lpil — COMMENTED** [review]
Thanks! The mix version can be ignored as it is only used for testing.

File: CHANGELOG.md (hunk 1 above)
Comment (williamthome) [reply CHANGELOG.md]: Can it be "Adds support for integer keys."?

**Review by lpil — APPROVED** [review]
Thank you! ✨

**Pull request merged**